lib/src/main/kotlin/starknet/types/Felt.kt

```diff
@@ -0,0 +1,33 @@
+package types
+
+import java.math.BigInteger
+
+data class Felt(val value: BigInteger) {
+    constructor(value: Long) : this(BigInteger.valueOf(value))
+
+    init {
+        if (value < BigInteger.ZERO) {
+            throw java.lang.IllegalArgumentException("Default Felt constructor does not accept negative numbers, [$value] given.")
+        }
+        if (value > PRIME) {
+            throw java.lang.IllegalArgumentException("Default Felt constructor does not accept numbers higher than P, [$value] given.")
+        }
+    }
+
+    companion object {
+        @field:JvmField
+        val PRIME = BigInteger("800000000000011000000000000000000000000000000000000000000000001", 16)
+
+        @field:JvmField
+        val ZERO = Felt(BigInteger.ZERO)
+
+        @field:JvmField
+        val ONE = Felt(BigInteger.ONE)
+
+        @JvmStatic
+        fun fromHex(value: String): Felt = Felt(BigInteger(value.removePrefix("0x"), 16))
```

Comment:
If we'd like to be more strict about hex vs dec strings, we could fail when the '0x' prefix is not provided. But I'm not sure about it.